cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs

```diff
@@ -70,6 +71,11 @@ instance IsString (Hash h a) where
 
 instance (HashAlgorithm h, Typeable a) => ToCBOR (Hash h a) where
   toCBOR = toCBOR . getHash
+  encodedSizeExpr _size proxy =
+      encodedSizeExpr (\_ -> hashSize) (getHash <$> proxy)
+    where
+      hashSize :: Size
+      hashSize = fromIntegral (sizeHash (Proxy :: Proxy h))
```

Comment:
I don't understand this. It needs explaining.

I presume what we're doing here is reusing the `ByteString` instance (that's what the `(getHash <$> proxy)` gives us) and then passing as the size expression the hash length.

If I'm right, that took me about 10min of studying the `encodedSizeExpr` code to figure out.

Reply:
Yes, it does exactly that :).